scripts-dev/mypy_synapse_plugin.py

```diff
@@ -153,6 +156,11 @@ def cached_function_method_signature(ctx: MethodSigContext) -> CallableType:
 
 
 def check_is_cacheable_wrapper(ctx: MethodSigContext) -> CallableType:
+    """Asserts that the signature of a method returns a value which can be cached.
+
+    Makes no changes to the provided method signature.
```

Comment:
Can we add a sentence explaining why there is a wrapper? It looks like the point is mainly to check that we have a proper `CallableType`(?)

OTOH if it's just "to have smaller functions" then let's leave it as is.